opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -82,20 +101,23 @@ impl LoggerProvider {
 
     /// Shuts down this `LoggerProvider`, panicking on failure.
     pub fn shutdown(&mut self) -> Vec<LogResult<()>> {
-        self.try_shutdown()
-            .expect("cannot shutdown LoggerProvider when child Loggers are still active")
+        // mark itself as already shutdown
+        self.is_shutdown
+            .store(true, std::sync::atomic::Ordering::Relaxed);
+        // propagate the shutdown signal to processors
+        // it's up to the processor to properly block new logs after shutdown
+        self.inner
+            .processors
+            .iter()
+            .map(|processor| processor.shutdown())
+            .collect()
     }
 
     /// Attempts to shutdown this `LoggerProvider`, succeeding only when
     /// all cloned `LoggerProvider` values have been dropped.
+    // todo: remove this
```

Comment:
nit - If we are keeping `try_shutdown()` for backward compatibility, good to update the existing comments, as now it doesn't just attempt, but really shutdown the LoggerProvider.